.github/workflows/r.yml

```diff
@@ -324,6 +324,10 @@ jobs:
           cd r/tests
           sed -i.bak -E -e 's/"arrow"/"arrow", reporter = "location"/' testthat.R
           rm -f testthat.R.bak
+      - name: Build cpp11 (on R 3.6)
+        if: ${{ matrix.rtools == 35 }}
+        shell: Rscript {0}
+        run: install.packages("cpp11", type = "source")
```

Comment:
Why does this matter? cpp11 is header-only so the "binary" package isn't meaningful I didn't think

Reply:
The headers changed, so when we go to build we get namespace errors:

```
ltrep.cpp: In static member function 'static arrow::r::altrep::{anonymous}::AltrepVectorPrimitive<sexp_type>::c_type arrow::r::altrep::{anonymous}::AltrepVectorPrimitive<sexp_type>::Elt(SEXP, R_xlen_t)':
576
altrep.cpp:245:31: error: 'na' is not a member of 'cpp11'
577
     return array->IsNull(j) ? cpp11::na<c_type>()
578
                               ^
579
altrep.cpp:245:47: error: expected primary-expression before '>' token
580
     return array->IsNull(j) ? cpp11::na<c_type>()
581
                                               ^
582
altrep.cpp:245:49: error: expected primary-expression before ')' token
583
     return array->IsNull(j) ? cpp11::na<c_type>()
584
                                                 ^
585
altrep.cpp: In static member function 'static R_xlen_t arrow::r::altrep::{anonymous}::AltrepVectorPrimitive<sexp_type>::Get_region(SEXP, R_xlen_t, R_xlen_t, arrow::r::altrep::{anonymous}::AltrepVectorPrimitive<sexp_type>::c_type*)':
586
altrep.cpp:283:22: error: 'na' is not a member of 'cpp11'
587
             out[j] = cpp11::na<c_type>();
588
                      ^
589
altrep.cpp:283:38: error: expected primary-expression before '>' token
590
             out[j] = cpp11::na<c_type>();
591
                                      ^
592
altrep.cpp:283:40: error: expected primary-expression before ')' token
593
             out[j] = cpp11::na<c_type>();
```


https://github.com/jonkeane/arrow/runs/4551051925?check_suite_focus=true#step:17:575

Reply:
I see, we install with `options(pkgType="win.binary")` so that's why on R 3.6 we weren't getting new enough cpp11.

Reply:
Yup yup. I looked around to see if there was some way to encourage `type="source"` in cases where it's *needed* but it's all or nothing as far as I've found.